lib/reservation_tracker.rb

```diff
@@ -0,0 +1,78 @@
+module Hotel
+class ReservationTracker
+
+  attr_reader :reservations, :blocks
+
+  def initialize
+    @reservations = []
+    @blocks = []
+  end
+
+  def list_rooms
+    return [*1..20]
+  end
+
+  def reservations_during_date(date)
+
+    valid_reservations = @reservations.find_all do |reservation|
+      reservation.date_within_reservation?(date)
+    end
+
+    return valid_reservations.empty? ? nil : valid_reservations
+  end
+
+  def available_rooms(start_date, end_date)
+
+    unavailable_rooms = []
+
+    @reservations.each do |reservation|
+      if reservation.daterange_within_reservation?(start_date, end_date)
+        unavailable_rooms << reservation.room
+      end
+    end
+
+    @blocks.each do |block|
+      if block.daterange_within_block?(start_date, end_date)
+        unavailable_rooms << reservation.room
+      end
+    end
+
+    return ([*1..20] - unavailable_rooms.uniq)
```

Comment:
Instead of using a magic number `[*1..20]`, you might want to call `list_rooms` here